src/io/parsers/CIFParser.js

```diff
@@ -60,753 +59,755 @@ function nameToElement(name) {
+      }
+      const name = names[i] || types[i];
+      const chain = complex.getChain(asymDict[chains[i]]);
 
-    // TODO think about last condition
-    if (!start || !end) {
-      continue;
-    }
-    var comment = comments[i] || '';
-    var length = lengths[i] || 0;
-    var helixClass = helixClasses[i] || ' ';
-    var struct;
-    // TODO Add turns and strands(!)?
-    if (type === 'helix') {
-      const idx = complex._helices.length;
-      struct = new Helix(helixClass, start[0], end[0], idx, name, comment, length);
-      complex.addHelix(struct);
-      complex.structures.push(struct);
-    } else if (type === 'turn') {
-      struct = new StructuralElement(StructuralElement.Type.TURN, start[0], end[0]);
-      complex.structures.push(struct);
-    } else {
-      struct = null;
-    }
-    if (!struct) {
-      continue;
-    }
-    var residues = chain.getResidues();
-    for (var r = start[1]; r <= end[1]; ++r) {
-      residues[r]._secondary = struct;
-    }
-  }
-};
+      const startIdx = starts[i];
+      const endIdx = ends[i];
+      const startICode = stICodes[i] || ' ';
+      const endICode = endICodes[i] || ' ';
 
-function _getOperations(operList) {
-  if (!operList) {
-    return null;
-  }
-  var idc = arrize(operList.id);
-  var matrix = operList.matrix;
-  var vector = operList.vector;
-  if (!idc || !matrix || !vector) {
-    return null;
-  }
+      const start = chain.findResidue(startIdx, startICode);
+      const end = chain.findResidue(endIdx, endICode);
 
-  var ops = [];
-  for (var i = 0, n = idc.length; i < n; ++i) {
-    var mtx = new THREE.Matrix4();
-    var elements = mtx.elements;
-
-    for (var row = 0; row < 3; ++row) {
-      var matrixData = matrix[row + 1];
-      elements[row]      = arrize(matrixData[1])[i];
-      elements[row + 4]  = arrize(matrixData[2])[i];
-      elements[row + 8]  = arrize(matrixData[3])[i];
-      elements[row + 12] = arrize(vector[row + 1])[i];
+      // TODO think about last condition
+      if (!start || !end) {
+        continue;
+      }
+      const comment = comments[i] || '';
+      const length = lengths[i] || 0;
+      const helixClass = helixClasses[i] || ' ';
+      let struct;
+      // TODO Add turns and strands(!)?
+      if (type === 'helix') {
+        const idx = complex._helices.length;
+        struct = new Helix(helixClass, start[0], end[0], idx, name, comment, length);
+        complex.addHelix(struct);
+        complex.structures.push(struct);
+      } else if (type === 'turn') {
+        struct = new StructuralElement(StructuralElement.Type.TURN, start[0], end[0]);
+        complex.structures.push(struct);
+      } else {
+        struct = null;
+      }
+      if (!struct) {
+        continue;
+      }
+      const residues = chain.getResidues();
+      for (let r = start[1]; r <= end[1]; ++r) {
+        residues[r]._secondary = struct;
+      }
     }
-    ops[idc[i]] = mtx;
   }
 
-  return ops;
-}
+  static _getOperations(operList) {
```

Comment:
Do we need this as a class member or should leave it be as a free function? Please check other such functions in this file and in others.